hasktorch/src/Torch/Functions.hs

```diff
@@ -90,9 +90,18 @@ sub a b = unsafePerformIO $ (cast3 ATen.sub_tts) a b kOne
 mul :: Tensor -> Tensor -> Tensor
 mul a b = unsafePerformIO $ (cast2 ATen.mul_tt) a b
 
+cadd :: Scalar a => Tensor -> a -> Tensor
+cadd t a = unsafePerformIO $ (cast2 ATen.add_ts) t a
```

Comment:
I'd rather avoid carrying over those weird names from TH. Why not `addScalar`?

Reply:
Alternatively why don't we make the signature of `add` this:
```hs
class TensorLike a where
  asTensorLike :: Tensor -> a -> Tensor -- first tensor used only for dtype/device

add :: TensorLike a => Tensor -> a -> Tensor
```

Reply:
Changed the library functions to use descriptive names, also backported some of the other examples to use descriptive function names.